LibraBFT/Impl/Handle.agda

```diff
@@ -32,157 +35,148 @@ open import LibraBFT.Yasm.Base
-      me                          = 0
-  Init.initialize me nfLiwsVssVvPe now ObmNeedFetch∙new pg
-
-initEMWithOutput : EitherD ErrLog (EpochManager × List Output)
-initEMWithOutput = do
-  (nf , _ , vss , vv , pe , liws) ← fromEither
-                                  $ GenKeyFile.create 1 (0 ∷ 1 ∷ 2 ∷ 3 ∷ [])
-  let nfLiwsVssVvPe               = (nf , liws , vss , vv , pe)
-      me                          = 0
-  fromEither $ Init.initialize me nfLiwsVssVvPe now ObmNeedFetch∙new pg
-
--- This shows that the Either and EitherD versions are equivalent.  This
--- is a first step towards eliminating the painful VariantOf stuff, so
--- we can have the version that looks (almost) exactly like the Haskell,
--- and the EitherD variant, broken into explicit steps, etc. for proving.
-initEMWithOutput≡ : initEMWithOutput' ≡ EitherD-run initEMWithOutput
-initEMWithOutput≡
-  with GenKeyFile.create 1 (0 ∷ 1 ∷ 2 ∷ 3 ∷ [])
-... | Left err = refl
-... | Right (nf , _ , vss , vv , pe , liws)
-  with Init.initialize 0 (nf , liws , vss , vv , pe) now ObmNeedFetch∙new pg
-... | Left err = refl
-... | Right y  = refl
+-- NOTE: The system layer only cares about this step function.
+-- 0 is given as a timestamp.
+peerStep : NodeId → NetworkMsg → RoundManager → RoundManager × List (LYT.Action NetworkMsg)
+peerStep nid msg st = runHandler st (handle nid msg 0)
+ where
+  -- This invokes an implementation handler.
+  runHandler : RoundManager → LBFT Unit → RoundManager × List (LYT.Action NetworkMsg)
+  runHandler st handler = ×-map₂ (outputsToActions {st}) (proj₂ (LBFT-run handler st))
 
 ------------------------------------------------------------------------------
--- TODO : ASK CHRIS : regarding EitherD-run
-
-zzz : EitherD ErrLog ℕ
-zzz = do
-  r ← RightD 0
-  RightD r
-
-zzz' : Either ErrLog ℕ
-zzz' = Right 0
 
-zzz≡zzz' : zzz ≡ RightD 0 → zzz' ≡ Right 0
-zzz≡zzz' ()
+-- This connects the implementation handler to the system model so it can be initialized.
+
+module InitHandler where
+  {-
+  IMPL-DIFF: In Haskell, nodes are started with a filepath of a file containing
+  - number of faults allowed
+  - genesis LedgerInfoWithSignatures
+  - network addresses/name and secret and public keys of all nodes in the genesis epoch
+
+  Main reads/checks that file then calls a network specific 'run' functions (e.g., ZMQ.hs)
+  that setup the network handlers then eventually call 'Start.startViaConsensusProvider'.
+
+  That functions calls 'ConsensusProvider.startConsensus' which returns
+  '(EpochManager, [Output]'.
+
+  'Start.startViaConsensusProvider' then goes on to create and wire up internal communication
+  channels and starts threads.  The most relevant thread starts up
+  `(EpochManager.obmStartLoop epochManager output ...)' to handle the initialization output
+  and then to handle new messages from the network.
+
+  In Agda functions below, assuming no initialization errors,
+  - the 'GenKeyFile.create' function
```

Comment:
I think these comments are out of date, as the Agda code does not create the file and read it to construct `BootStrapInfo`; we simply assume `bootStrapInfo` known to everyone.  I think it's good to keep the comments about how the Haskell implementation initialises, but also to be accurate about how we're approaching it in Agda.

Reply:
DONE